### PR TITLE
API client fixes

### PR DESCRIPTION
**Pull request by teneightfive**
### What changed

Includes two fixes the API client:

- don't store `undefined` in cache keys
- include querystring in URL during debug logging

### Why did it change

The cache keys should only return params that are sent as part of the URL.

Exposing the full URL allows us to see the differences in calls to the API. In some cases it differs on querystring filters like `page` so its useful to be able to distinguish those requests.

#### Automated testing

- [x] Added unit tests to cover changes
- [ ] ~Added end-to-end tests to cover any journey changes~

#### Manual testing

Has been tested in the following browsers:

- [x] Chrome
- [x] Firefox
- [ ] Edge
- [ ] Internet Explorer

### Environment variables

- [x] No environment variables were added or changed

### Other considerations

Commit messages with a `fix` or `feat` type are automatically used to generate the [changelog](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/CHANGELOG.md) and
[GitHub release notes](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/releases) during the release task. Please make sure they will read well on their own in a
summary of changes and that the commit body gives a more detailed description of those changes if necessary.

- [x] No new [Personally Identifiable Information (PII)](https://support.google.com/analytics/answer/6366371?hl=en) is being sent via analytics
- [x] Update [README](ministryofjustice/hmpps-book-secure-move-frontend/blob/master/README.md) with any new instructions or tasks

### CODE REVIEW

File: common/lib/api-client/middleware/request.js [hunk 1]
```diff
@@ -13,8 +13,10 @@ function requestMiddleware({ cacheExpiry = 60, useRedisCache = false } = {}) {
       }
 
       const { req, jsonApi, cacheKey } = payload
+      const searchString = new URLSearchParams(req.params).toString()
```
Comment (Thource) [inline common/lib/api-client/middleware/request.js:16]: Does this want to use pickBy too?

Comment (teneightfive) [reply common/lib/api-client/middleware/request.js:16]: Interesting point actually. The request middleware is actually after the [`requestInclude`](https://github.com/ministryofjustice/hmpps-book-secure-move-frontend/blob/2181f093f537546a9b04ce4a8dc1a05cb00a9aa1/common/lib/api-client/middleware/request-include.js#L22-L26) middleware that cleans up that key.

I wonder if we actually want to have something at the start of the middleware that cleans the params of any undefined values... @solidgoldpig what are your thoughts as I think you added the request include middleware?

**Comment by teneightfive** [comment]
@martyn-w this should solve the `include=undefined` you spotted in the redis cache.

File: common/lib/api-client/middleware/request.js (hunk 1 above)
Comment (solidgoldpig) [reply common/lib/api-client/middleware/request.js:16]: axios strips them out. Do it again here if you want ;)

**Review by solidgoldpig — APPROVED** [review]

**Pull request merged**